staging/src/k8s.io/apiserver/pkg/util/shufflesharding/shufflesharding.go

```diff
@@ -18,20 +18,34 @@ package shufflesharding
 
 import (
 	"errors"
+	"fmt"
 	"math"
+	"strings"
 )
 
 const maxHashBits = 60
 
-// ValidateParameters can validate parameters for shuffle sharding
-// in a fast but approximate way, including deckSize and handSize
-// Algorithm: maxHashBits >= bits(deckSize^handSize)
-func ValidateParameters(deckSize, handSize int32) bool {
-	if handSize <= 0 || deckSize <= 0 || handSize > deckSize {
-		return false
+// ValidateParameters finds errors in the parameters for shuffle
+// sharding.  Returns an empty slice iff there are no errors.  The
```

Comment:
is it an empty slice or nil slice returned when there are no errors?

Reply:
The `nil` slice is empty.  https://play.golang.org/p/AdlgjTL2DdP

Reply:
Oh, by "empty" I meant "len() returns 0".  I will reword to make it clearer.